### PR TITLE
Adding knapsack problem for benchmarking

### PR DESCRIPTION
**Pull request by alejomonbar**
### Summary

I add the knapsack problem for benchmarking, I used the same syntaxis of the Maxcut problem.

### CODE REVIEW

File: optimization/benchmarks/knapsack.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+# pylint: disable=redefined-outer-name, invalid-name, attribute-defined-outside-init
+
+
+class KnapsackBenchmarks:
+    """Knapsack benchmarks"""
+
+    num_items = [2, 4, 8, 12]
+
+    params = ([2, 4, 8, 16], [8, 16, 32, 64])
```
Comment (t-imamichi) [inline optimization/benchmarks/knapsack.py:35]: Could you adjust param numbers so that each benchmark takes at most a couple of seconds?

**Comment by t-imamichi** [comment]
I have two comments
1. File permission seems 755 (executable). can you change it to 644 (read)?
2. I recommend you to make another branch (not main) when you make a PR in the future. You don't have to change branch for this PR.

**Comment by manoelmarques** [comment]
I run the benchmarks in "quick mode" in the CI. Even though `asv` passes it is good to always take a look at the output because it seems the knapsack code fails:
```
File "/home/runner/work/qiskit-app-benchmarks/qiskit-app-benchmarks/optimization/benchmarks/knapsack.py", line 86, in time_grover
   meo.solve(self._qp)
 File "/home/runner/work/qiskit-app-benchmarks/qiskit-app-benchmarks/optimization/.asv/env/b84e110a83fb6d276d9e803f64f828f3/lib/python3.8/site-packages/qiskit_optimization/algorithms/grover_optimizer.py", line 235, in solve
   outcome = self._measure(circuit)
 File "/home/runner/work/qiskit-app-benchmarks/qiskit-app-benchmarks/optimization/.asv/env/b84e110a83fb6d276d9e803f64f828f3/lib/python3.8/site-packages/qiskit_optimization/algorithms/grover_optimizer.py", line 317, in _measure
   probs = self._get_probs(circuit)
 File "/home/runner/work/qiskit-app-benchmarks/qiskit-app-benchmarks/optimization/.asv/env/b84e110a83fb6d276d9e803f64f828f3/lib/python3.8/site-packages/qiskit_optimization/algorithms/grover_optimizer.py", line 325, in _get_probs
   result = self.quantum_instance.execute(qc)
 File "/home/runner/work/qiskit-app-benchmarks/qiskit-app-benchmarks/optimization/.asv/env/b84e110a83fb6d276d9e803f64f828f3/lib/python3.8/site-packages/qiskit/utils/quantum_instance.py", line 682, in execute
   run_circuits(
 File "/home/runner/work/qiskit-app-benchmarks/qiskit-app-benchmarks/optimization/.asv/env/b84e110a83fb6d276d9e803f64f828f3/lib/python3.8/site-packages/qiskit/utils/run_circuits.py", line 622, in run_circuits
   raise QiskitError(f"Circuit execution failed: {msg}")
qiskit.exceptions.QiskitError: 'Circuit execution failed: ERROR: Failed to load qobj: AerSimulator: circuit with instructions {"instructions": {gate, measure}, "gates": {u2, mcphase, cp, h}} is not compatible with any available simulation methods'

For parameters: 16, 16
Simulation failed and returned the following error message:
ERROR: Failed to load qobj: AerSimulator: circuit with instructions {"instructions": {gate, measure}, "gates": {u2, mcphase, cp, h}} is not compatible with any available simulation methods
```

File: optimization/benchmarks/knapsack.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    def setup(self, num_items, max_weights):
+        """setup"""
+        seed = 123
+        algorithm_globals.random_seed = seed
+        qasm_sim = Aer.get_backend("aer_simulator")
+        self._qins = QuantumInstance(
+            backend=qasm_sim, shots=1, seed_simulator=seed, seed_transpiler=seed
+        )
+        values = [random.randint(1, max_weights) for i in range(num_items)]
+        weights = [random.randint(1, max_weights) for i in range(num_items)]
```
Comment (t-imamichi) [inline optimization/benchmarks/knapsack.py:46]:
```suggestion
        values = [random.randint(1, max_weights) for _ in range(num_items)]
        weights = [random.randint(1, max_weights) for _ in range(num_items)]
```

File: optimization/benchmarks/knapsack.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            min_eigen_solver=VQE(
+                optimizer=COBYLA(maxiter=1), ansatz=EfficientSU2(), quantum_instance=self._qins
+            )
+        )
+        meo.solve(self._qp)
+
+    def time_grover(self, _, __):
+        """time grover"""
+        meo = GroverOptimizer(
+            num_value_qubits=self._qp.get_num_vars() // 2,
```
Comment (t-imamichi) [inline optimization/benchmarks/knapsack.py:80]:
```suggestion
            num_value_qubits=self._qp.get_num_vars(),
```

Comment (alejomonbar) [reply optimization/benchmarks/knapsack.py:80]: Thank you!

**Review by t-imamichi — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**